src/java.base/share/classes/java/lang/StableValue.java

```diff
@@ -0,0 +1,746 @@
+ *           (directly or indirectly) synchronizing on a {@code StableValue}. Failure to
+ *           do this may lead to deadlock. Stable functions and collections on the
+ *           other hand are guaranteed <em>not to synchronize</em> on {@code this}.
+ *           Except for a {@code StableValue}'s content itself, an {@linkplain #orElse(Object) orElse(other)}
+ *           parameter, and an {@linkplain #equals(Object) equals(obj)} parameter; all
+ *           method parameters must be <em>non-null</em> or a {@link NullPointerException}
+ *           will be thrown.
+ *
+ * @implNote A {@linkplain StableValue} is mainly intended to be a non-public field in
+ *           a class and is usually neither exposed directly via accessors nor passed as
+ *           a method parameter.
+ *           Stable functions and collections make all reasonable efforts to provide
+ *           {@link Object#toString()} operations that do not trigger evaluation
+ *           of the internal stable values when called.
+ *           Stable collections have {@link Object#equals(Object)} operations that tries
+ *           to minimize evaluation of the internal stable values when called.
+ *           As objects can be set via stable values but never removed, this can be a source
+ *           of unintended memory leaks. A stable value's content is
+ *           {@linkplain java.lang.ref##reachability strongly reachable}. Clients are
+ *           advised that {@linkplain java.lang.ref##reachability reachable} stable values
+ *           will hold their set content perpetually.
+ *           A {@linkplain StableValue} that has a type parameter {@code T} that is an array
+ *           type (of arbitrary rank) will only allow the JVM to treat the <em>array reference</em>
+ *           as a stable value but <em>not its components</em>. Clients can instead use
+ *           {@linkplain #list(int, IntFunction) a stable list} of arbitrary depth, which
+ *           provides stable components. More generally, a stable value can hold other
+ *           stable values of arbitrary depth and still provide transitive constantness.
+ *
+ * @param <T> type of the content
+ *
+ * @since 25
+ */
+@PreviewFeature(feature = PreviewFeature.Feature.STABLE_VALUES)
+public sealed interface StableValue<T>
+        permits StableValueImpl {
+
+    // Principal methods
+
+    /**
+     * Tries to set the content of this StableValue to the provided {@code content}.
+     * <p>
+     * When this method returns, the content of this StableValue is always set.
+     *
+     * @return {@code true} if the content of this StableValue was set to the
+     *         provided {@code content}, {@code false} otherwise
+     * @param content to set
+     * @throws IllegalStateException if this method is invoked directly by a supplier
+     *         provided to the {@link #orElseSet(Supplier)} method.
+     */
+    boolean trySet(T content);
```

Comment:
I assume the intent is that trySet can only return at most true once but the current wording suggests that two threads racing to set the same content would return true, or sequential calls to trySet to set the same content would return true. Does this need to be clarified?

Reply:
So maybe "was set to the provided content" -> "was set by this call"